### PR TITLE
chore(auth): Update sentry.identity.oauth2 to support customer domains

### PR DESCRIPTION
**Pull request by dashed**
Branches from https://github.com/getsentry/sentry/pull/39016.

This delegates the pipeline redis store to propagate customer domain information for the OAuth2 implementation that resides in [`sentry.identity.oauth2`](https://github.com/getsentry/sentry/blob/master/src/sentry/identity/oauth2.py).

### CODE REVIEW

File: tests/sentry/identity/test_oauth2.py [hunk 1]
```diff
@@ -30,17 +38,49 @@ def view(self):
             client_secret="secret-value",
         )
 
-    @responses.activate
```
Comment (markstory) [inline tests/sentry/identity/test_oauth2.py:33]: Was responses not able to do the mocking you needed? It should be able to mock out anything using urllib3.

Comment (dashed) [reply tests/sentry/identity/test_oauth2.py:33]: @markstory I added asserts to check the arguments passed to `safe_urlopen()` for the token exchange.

Comment (markstory) [reply tests/sentry/identity/test_oauth2.py:33]: You can do that with responses as well. `responses.calls` gives you access to all the requests that were made and the url/parameters that were used for each mocked request.

**Comment by armenzg** [comment]
Untagging ecosystem. Retag if our review is needed.

File: tests/sentry/identity/test_oauth2.py (hunk 1 above)
Comment (dashed) [reply tests/sentry/identity/test_oauth2.py]: @markstory I've updated to `responses.calls`.

**Review by maxiuyuan — APPROVED** [review]

File: src/sentry/integrations/pipeline.py [hunk 2]
```diff
@@ -177,3 +178,19 @@ def _dialog_success(self, org_integration):
...
+        view = self.pipeline_views[step_index]
+        if isinstance(view, NestedPipelineView):
+            # Attempt to surface state from a nested pipeline
+            nested_pipeline = view.pipeline_cls(
+                organization=self.organization,
+                request=self.request,
+                provider_key=view.provider_key,
+                config=view.config,
+            )
+            return nested_pipeline.fetch_state(key)
```
Comment (evanpurkhiser) [inline src/sentry/integrations/pipeline.py:195]: Why didn’t you just put this in the base pipeline fetch_state?

Comment (dashed) [reply src/sentry/integrations/pipeline.py:195]: @evanpurkhiser https://github.com/getsentry/sentry/pull/39689

**Pull request merged**